pymc3/sampling.py

```diff
@@ -1283,8 +1322,9 @@ def sample_prior_predictive(samples=500, model=None, vars=None, random_seed=None
     values = draw_values([model[name] for name in names], size=samples)
 
     data = {k: v for k, v in zip(names, values)}
+    assert data is not None
```

Comment:
raw `assert`s are considered bad form (they can be ignored if you use some flag that no one actually uses) -- `if data is None: raise AssertionError`

Reply:
Fixed in 65439179b1abe7d8e14b2dbf01d257034f669e7e

Reply:
OK, I have no idea why this happened: when I replaced that line per above I now get a bunch of test failures that look like this:

```
pymc3/tests/test_distributions.py:500: in check_dlogp
    assert_almost_equal(dlogp(pt), ndlogp(pt), decimal=decimals, err_msg=str(pt))
../../../miniconda3/envs/testenv/lib/python3.6/site-packages/numdifftools/core.py:723: in __call__
    *args, **kwds).squeeze()
../../../miniconda3/envs/testenv/lib/python3.6/site-packages/numdifftools/core.py:666: in __call__
    return super(Jacobian, self).__call__(np.atleast_1d(x), *args, **kwds)
../../../miniconda3/envs/testenv/lib/python3.6/site-packages/numdifftools/core.py:297: in __call__
    results = self._derivative(xi, args, kwds)
../../../miniconda3/envs/testenv/lib/python3.6/site-packages/numdifftools/core.py:663: in _derivative_nonzero_order
    return self._apply_fd_rule(step_ratio, results, steps2)
../../../miniconda3/envs/testenv/lib/python3.6/site-packages/numdifftools/core.py:394: in _apply_fd_rule
    fd_rule = self._get_finite_difference_rule(step_ratio)
../../../miniconda3/envs/testenv/lib/python3.6/site-packages/numdifftools/core.py:378: in _get_finite_difference_rule
    fd_mat = self._fd_matrix(step_ratio, parity, num_terms)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
step_ratio = 2.0, parity = 1, nterms = 1
    @staticmethod
    def _fd_matrix(step_ratio, parity, nterms):
        """
        Return matrix for finite difference and complex step derivation.
    
        Parameters
        ----------
        step_ratio : real scalar
            ratio between steps in unequally spaced difference rule.
        parity : scalar, integer
            0 (one sided, all terms included but zeroth order)
            1 (only odd terms included)
            2 (only even terms included)
            3 (only every 4'th order terms included starting from order 2)
            4 (only every 4'th order terms included starting from order 4)
            5 (only every 4'th order terms included starting from order 1)
            6 (only every 4'th order terms included starting from order 3)
        nterms : scalar, integer
            number of terms
        """
        _assert(0 <= parity <= 6,
                'Parity must be 0, 1, 2, 3, 4, 5 or 6! ({0:d})'.format(parity))
        step = [1, 2, 2, 4, 4, 4, 4][parity]
        inv_sr = 1.0 / step_ratio
        offset = [1, 1, 2, 2, 4, 1, 3][parity]
        c0 = [1.0, 1.0, 1.0, 2.0, 24.0, 1.0, 6.0][parity]
        c = c0 / \
>           misc.factorial(np.arange(offset, step * nterms + offset, step))
E       AttributeError: module 'scipy.misc' has no attribute 'factorial'
../../../miniconda3/envs/testenv/lib/python3.6/site-packages/numdifftools/core.py:330: AttributeError
```

I do **not** see these on my laptop -- all the tests pass there.  This does not seem to have anything to do with my change to the `assert`.

Reply:
my guess is a version of something (`numdifftools`?) changed under this diff. Comparing versions from the previous travis build and this one should let us know (and figure out if there's a good change). I'll dig into it after work.

Reply:
That looks right -- I don't think it's anything I did. Is there some way to force a re-run of Travis on `master`, so that we can see if this failure also happens there (and my little tweak is innocent)?

Reply:
@ColCarroll `numdifftools` didn't change, **but**  `scipy` 1.3.0 was released three days ago (17 May).

And yes,`factorial` was removed from `scipy.misc`: https://github.com/scipy/scipy/commit/1cde30518fdc46c2c765beaa3e132e322ea7a579#diff-0763dd201fcbf9b088f9eb502b4f3d42

When you've got the factorial problem fixed, please squash merge this PR, or LMK if you would prefer me to smash it down to one commit for you to merge. Thanks!